postrelease-vip.php

```diff
@@ -2,567 +2,680 @@
  */
- 
-/*
-+-----------------------------------------------------------------------------------------
-| Version | Date       | Notes
-|---------|-------------------------------------------------------------------------------
-| 1.0     | 2012/08/01 | First release live
-| 1.1     | 2012/10/05 | Returning blog URL as the blog name as fallback
-| 1.2     | 2012/10/08 | Added filters for date and author (display as Promoted)
-+-----------------------------------------------------------------------------------------
-*/
- 
-define('PRX_CUSTOM_TYPE', 'pr_sponsored_post'); //custom post type created for our template post 
-define('PRX_WP_PLUGIN_VERSION', '1.2');
-define('PRX_DB_VERSION', 1);
-
-require_once( dirname(__FILE__) .'/config.php' );
-
-add_action('init', 'postrelease_init');
-add_action('init', 'postrelease_handle_enable_request' );
-add_action('init', 'postrelease_handle_signup_requests');
-add_action('admin_init', 'postrelease_admin_init');
-add_action('wp_enqueue_scripts', 'postrelease_wp_enqueue_scripts', 1);
-add_action('admin_menu', 'postrelease_create_menu'); // create custom plugin settings menu
-add_filter('query_vars', 'postrelease_add_query_vars');
-add_action('template_redirect', 'postrelease_template_redirect');
-add_action('the_author', 'postrelease_the_author');
-add_filter('author_link', 'postrelease_author_link', 100, 2);
-add_action('the_author_display_name', 'postrelease_the_author_display_name');
-add_filter('get_the_date', 'postrelease_get_the_date', 10, 2);
-add_filter('get_the_time', 'postrelease_get_the_time', 10, 2);
-
-if (!function_exists('wpcom_is_vip')) {
-	register_activation_hook( __FILE__, 'postrelease_notify_activate' );
-	register_deactivation_hook( __FILE__, 'postrelease_notify_deactivate' );	
-}
-
-/**
- * Appends the PostRelease javascript to header 
- */
-function postrelease_wp_enqueue_scripts() {
-	//if plugin is activated and we are not in an admin page
-	if ( 1 == get_option( 'prx_plugin_activated', 0 ) && ! is_admin() ) {
-	    wp_register_script('postrelease', PRX_JAVASCRIPT_URL);
-	    wp_enqueue_script('postrelease');
-	}
-}
-
-function postrelease_init() {
-	$args = array('public' => false || PRX_DEV,
-				  'publicly_queryable' => true,
-	              'label' => 'Sponsored Post',
-	    		  'exclude_from_search' => true,
-	    		  'can_export' => false,	 
-	    		  'supports' => array( 'title', 'editor', 'author', 'thumbnail', 'excerpt' )
-	);
-	register_post_type( PRX_CUSTOM_TYPE, $args );
-}
-
-/**
- * Function when plugin is activated.
- * In WP VIP, the activate hook is not supported.
- * The plugin is activated when the user goes to the dashboard and
- * adds his publication to the PostRelease network. 
- */
-function postrelease_activate() {
-	if ( ! function_exists( 'wpcom_is_vip' ) ) {
-		flush_rewrite_rules(); //not vip
-	} else {
-		do_action( 'postrelease_activate' ); //vip
-	}
-}
+if ( ! class_exists( PostRelease_VIP ) ) :
+
+	class PostRelease_VIP {
```

Comment:
I don't see a need for `_VIP`. I would also say that the class (and plugin in general) should be renamed `Nativo` since that's the product's new name.